### PR TITLE
Renaming

### PR DESCRIPTION
**Pull request by marche0133**
Azure review for GA:

- Use same terminology than calling (Participant instead of Member)
- Use UpdateTopic instead of UpdateThread in operation name
- Use ChatMessageReadReceipt instead of ReadReceipt in model name
- Return messageId instead of SendMessageResult. (SendMessage operation)
- Add convenience method to add 1 member

All details : https://skype.visualstudio.com/DefaultCollection/SPOOL/_workitems/edit/2235800

### CODE REVIEW

File: sdk/communication/azure-communication-chat/README.md [hunk 1]
```diff
@@ -79,11 +81,11 @@ AzureCommunicationChatServiceAsyncClient client = new AzureCommunicationChatServ
 
 User access tokens enable you to build client applications that directly authenticate to Azure Communication Services. Refer [here](https://docs.microsoft.com/azure/communication-services/quickstarts/access-tokens) to learn how to create a user and issue a User Access Token.
 
-The id for the user created above will be necessary later to add said user as a member of a new chat thread. The initiator of the create request must be in the list of members of the chat thread.
+The id for the user created above will be necessary later to add said user as a participant of a new chat thread. The initiator of the create request must be in the list of participants of the chat thread.
```
Comment (juancamilor) [inline sdk/communication/azure-communication-chat/README.md:84]: This part is not true anymore : 'The initiator of the create request must be in the list of participants of the chat thread.'
Lets start removing that from comments and docs with the changes we are doing.

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java [hunk 2]
```diff
@@ -214,124 +141,101 @@ public void sendTypingNotification(String chatThreadId, final Callback<Void> cal
...
-     * @param callback the Callback that receives the response.
+     * @param chatThreadId Thread id to get participants for.
+     * @param maxPageSize The maximum number of participants to be returned per page.
+     * @param skip Skips participants up to a specified position in response.
+     * @param callback the Callback that receives the response collection.
      * @throws IllegalArgumentException thrown if parameters fail the validation.
      * @throws ErrorException thrown if the request is rejected by server.
      * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
      */
-    public void listChatThreadMembers(String chatThreadId, final Callback<Page<ChatThreadMember>> callback) {
```
Comment (vcolin7) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java:225]: Are we just removing all listing methods in favor of ones that return a paged collection?

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java [hunk 3]
```diff
@@ -214,124 +141,101 @@ public void sendTypingNotification(String chatThreadId, final Callback<Void> cal
...
      *
+     * @param createChatThreadRequest Request payload for creating a chat thread.
      * @param callback the Callback that receives the response.
      * @throws IllegalArgumentException thrown if parameters fail the validation.
      * @throws ErrorException thrown if the request is rejected by server.
      * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
      */
-    public void listChatThreads(final Callback<Page<ChatThreadInfo>> callback) {
-        this.serviceClient.listChatThreads(callback);
+    public void createChatThread(CreateChatThreadRequest createChatThreadRequest, final Callback<ChatThread> callback) {
```
Comment (vcolin7) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java:209]: nit; I would put the simpler overloads first, helps with readability.

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceClient.java [hunk 4]
```diff
@@ -61,90 +39,66 @@
...
-     * 
+     *
      * @param chatThreadId Thread id to send the read receipt event to.
-     * @param body Request payload for sending a read receipt.
+     * @param sendReadReceiptRequest Request payload for sending a read receipt.
      * @throws IllegalArgumentException thrown if parameters fail the validation.
      * @throws ErrorException thrown if the request is rejected by server.
      * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
-     * @return the response.
+     * @return the completion.
```
Comment (vcolin7) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceClient.java:79]: I don't think this is very descriptive. Maybe we could use something like "An empty response indicating the operation's completion."

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceClient.java [hunk 5]
```diff
@@ -61,90 +39,66 @@
...
      * Sends a message to a thread.
-     * 
+     *
      * @param chatThreadId The thread id to send the message to.
-     * @param body Details of the message to send.
+     * @param sendChatMessageRequest Details of the message to send.
      * @throws IllegalArgumentException thrown if parameters fail the validation.
      * @throws ErrorException thrown if the request is rejected by server.
      * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
      * @return result of the send message operation.
```
Comment (vcolin7) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceClient.java:93]: The return Javadoc should read "The ID of the message sent." or something like that.

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/AzureCommunicationChatServiceImpl.java [hunk 6]
```diff
@@ -4,24 +4,8 @@
...
-import com.azure.android.communication.chat.models.MultiStatusResponse;
-import com.azure.android.communication.chat.models.ReadReceipt;
-import com.azure.android.communication.chat.models.ReadReceiptsCollection;
-import com.azure.android.communication.chat.models.SendChatMessageRequest;
-import com.azure.android.communication.chat.models.SendChatMessageResult;
-import com.azure.android.communication.chat.models.SendReadReceiptRequest;
-import com.azure.android.communication.chat.models.UpdateChatMessageRequest;
-import com.azure.android.communication.chat.models.UpdateChatThreadRequest;
+import com.azure.android.communication.chat.implementation.converters.SendChatMessageResultConverter;
+import com.azure.android.communication.chat.models.*;
```
Comment (vcolin7) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/AzureCommunicationChatServiceImpl.java:8]: We probably want to use explicit imports.

File: sdk/communication/azure-communication-chat/README.md [hunk 7]
```diff
@@ -15,7 +15,9 @@ This package contains the Chat client library for Azure Communication Services.
 * The library is written in Java 8. Your application must be built with Android Gradle plugin 3.0.0 or later, and must be configured to [enable Java 8 language desugaring](https://developer.android.com/studio/write/java8-support.html#supported_features) to use this library. Java 8 language features that require a target API level > 21 are not used, nor are any Java 8+ APIs that would require the Java 8+ API desugaring provided by Android Gradle plugin 4.0.0.
 
 ### Versions available
-The current version of this library is **1.0.0-beta.2**.
+The current Server - Chat Api Version is **2020-11-01-preview3**.
+
+The current Client - Chat SDK Version is **1.0.0-beta.3**.
```
Comment (vcolin7) [inline sdk/communication/azure-communication-chat/README.md:20]: Maybe we can rephrase this to:
```suggestion
The current Azure Communication Chat Service Version is **2020-11-01-preview3**.

The current Azure Communication Chat SDK Version is **1.0.0-beta.3**.
```
Or some other way that makes it easier to read.

**Review by vcolin7 — CHANGES_REQUESTED** [review]
Some of my comments on a client might apply to its async/sync counterpart. Also, were these changes made manually or were they re-generated from an updated Swagger?

**Comment by JianpingChen** [comment]
Should this be merged into a feature branch (instead of master)? Feature branch seems to be the approach other languages are taking

**Comment by juancamilor** [comment]
I think it would be better to do it in a feature branch as well (all - including swift now) are doing it that way @marche0133

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java [hunk 8]
```diff
@@ -61,100 +27,61 @@
...
-     * Gets read receipts for a thread.
+     * Gets chat message read receipts for a thread.
      *
-     * @param chatThreadId Thread id to get the read receipts for.
-     * @param callback the Callback that receives the response.
-     * @throws IllegalArgumentException thrown if parameters fail the validation.
-     * @throws ErrorException thrown if the request is rejected by server.
-     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
-     */
-    public void listChatReadReceipts(String chatThreadId, final Callback<Page<ReadReceipt>> callback) {
```
[the remark this answers is not in the document]
Comment (JianpingChen) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java:72]: We do not want to expose "naked" page but only expose either PagedDataCollection or PageResponseDataCollection

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java [hunk 9]
```diff
@@ -61,100 +27,61 @@
...
-     *
-     * @param chatThreadId The thread id of the message.
-     * @param maxPageSize The maximum number of messages to be returned per page.
-     * @param startTime The earliest point in time to get messages up to. The timestamp should be in ISO8601 format: `yyyy-MM-ddTHH:mm:ssZ`.
-     * @param callback the Callback that receives the response.
-     * @throws IllegalArgumentException thrown if parameters fail the validation.
-     * @throws ErrorException thrown if the request is rejected by server.
-     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
-     */
-    public void listChatMessages(String chatThreadId, Integer maxPageSize, OffsetDateTime startTime, final Callback<Page<ChatMessage>> callback) {
```
[the remark this answers is not in the document]
Comment (JianpingChen) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java:128]: Similar idea as for ReadReceipts, the less we expose the lower we lose backward compatibility

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java (hunk 2 above)
Comment (JianpingChen) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java:225]: Yes. Anu and I think that's easier to maintain backward compatibility

**Review by JianpingChen — APPROVED** [review]

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (marche0133) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java:72]: > Some of my comments on a client might apply to its async/sync counterpart. Also, were these changes made manually or were they re-generated from an updated Swagger?

Most comments you list here are generated code from Swagger, the list message methods are removed by new autorest generator, Jianping did it intentionally to force user use Page methods.

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java [hunk 10]
```diff
@@ -5,47 +5,13 @@
...
-import com.azure.android.communication.chat.models.ErrorException;
-import com.azure.android.communication.chat.models.MultiStatusResponse;
-import com.azure.android.communication.chat.models.ReadReceipt;
-import com.azure.android.communication.chat.models.ReadReceiptsCollection;
-import com.azure.android.communication.chat.models.SendChatMessageRequest;
-import com.azure.android.communication.chat.models.SendChatMessageResult;
-import com.azure.android.communication.chat.models.SendReadReceiptRequest;
-import com.azure.android.communication.chat.models.UpdateChatMessageRequest;
-import com.azure.android.communication.chat.models.UpdateChatThreadRequest;
+import com.azure.android.communication.chat.models.*;
```
[the remark this answers is not in the document]
Comment (vcolin7) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java:8]: Does this mean the "import *" statements coming from AutoRest?

**Review by vcolin7 — APPROVED** [review]
